### ISSUE TITLE
Umbrella: Autoreason integration + productness CLS branch

### ISSUE THREAD

**Issue by joesu-angible** [joesu-angible/autoresearch#6]
## Purpose
Umbrella tracker for the next implementation wave in `joesu-angible/autoresearch`: integrate the Autoreason-style experiment/search loop into the current DINO teacher → LCNet student fine-tuning workflow, and add a binary product-vs-personal-item CLS branch for deployment filtering.

## Sub-issues

- #4 — Integrate Autoreason-style experiment tournament into DINO teacher → student autoresearch workflow
- #5 — Add product-vs-personal-item CLS branch to student ReID model

### 1. Autoreason / tournament integration

Use Autoreason-like search as an outer-loop experiment controller around the existing pipeline, not as a replacement for the training stack:

1. train / select stronger DINO teacher(s);
2. distill into LCNet student via `student_finetune`;
3. evaluate with retrieval + teacher-alignment metrics;
4. run tournament-style candidate selection over configs, losses, teachers, and sampling strategies;
5. preserve full experiment records so future runs can build on previous winners.

### 2. Productness CLS branch

Add a small auxiliary binary classifier on top of LCNet shared features:

```text
1 = 商品 / product
0 = 個人物品 / personal item / blacklist negative
```

Initial design direction:

- keep `encode()` embedding-only and backward compatible;
- keep retrieval metrics separate from productness metrics;
- use BCE-with-logits with a conservative initial weight (`0.01–0.05`, suggested start `0.02`);
- add optional ONNX output `(embedding, productness_score)` without changing default embedding-only export.

## Suggested work branch

`feat/autoreason-cls-umbrella`

## Acceptance criteria

- Existing student fine-tune path still works without the new CLS branch enabled.
- Productness labels are explicitly plumbed from dataset → collate → training loop.
- Productness loss/accuracy are logged separately.
- Retrieval validation is not polluted by personal-item negatives.
- ONNX embedding-only export remains backward compatible.
- Autoreason/tournament loop can run as an outer orchestration layer and produce reproducible candidate records.

## Notes

This umbrella issue should stay high-level. Implementation details and risk discussion belong in the child issues above.
